src/hooks/useIntervalWhen.ts

```diff
@@ -25,6 +27,9 @@ function useIntervalWhen(
 
   useEffect(() => {
     if (when) {
+      if (eager) {
+        callback();
```

Comment:
Won't this run even before the timeout? This will run well before the interval. Am I missing something?

Reply:
What do you think of this @qiweiii ?

Reply:
@imbhargav5 Yes, it runs the callback before timer starts.

I think @mar1u50 means when we start polling a backend service, we usually don't want to wait for **the first interval**, we want to send out the first request right away, then start the timer and wait for a while and send another request and repeat. I think it is a common scenario. Good point @mar1u50!

Reply:
Ah. Sounds good. I like the use case. Let's document this option in the Readme, add an example and finish this off @qiweiii